### PR TITLE
Issue #649: Wrong days remaining label when submitting a new proposal

### PR DESCRIPTION
**Pull request by raluka**

### CODE REVIEW

File: app/views/proposal/index.html.haml [hunk 1]
```diff
@@ -97,5 +97,5 @@
                   class: 'btn btn-default', id: "edit_proposal_#{event.id}"
   .row
     .col-md-12
-      - if @conference.cfp_open? || (current_user.has_role? :organizer, @conference)
+      - if @conference.cfp_open?
```
Comment (ChrisBr) [inline app/views/proposal/index.html.haml]: Why do you remove
```
|| (current_user.has_role? :organizer, @conference)
```

File: app/views/admin/call_for_papers/show.html.haml [hunk 2]
```diff
@@ -19,8 +19,12 @@
         %dt
           Days Left:
         %dd
-          - days = (@call_for_paper.end_date - Date.today).to_i
+          - if @conference.cfp_open?
```
Comment (ChrisBr) [inline app/views/admin/call_for_papers/show.html.haml]: Hm, maybe we should move this to a helper or model method?

**Comment by ChrisBr** [comment]
Except my two comments it LGTM!
Thanks @raluka

File: app/views/proposal/index.html.haml (hunk 1 above)
Comment (raluka) [inline app/views/proposal/index.html.haml]: With that condition, if you have role of organizer for conference, you can still see "New Proposal" button if call_for_papers period is closed. Also, you can submit new proposal after call_for_papers period is closed.

Comment (ChrisBr) [inline app/views/proposal/index.html.haml]: I know! But as an Organizer I want to be able to submit proposals after the CFP is closed. Not sure if we already cover this in the admin menu.

@hennevogel @differentreality what do you think?

Comment (raluka) [inline app/views/proposal/index.html.haml]: ok

**Comment by raluka** [comment]
@ChrisBr , if everything is ok here, please let me know, so I squash commits. Thanks

File: app/views/conference/_call_for_paper.html.haml [hunk 3]
```diff
@@ -25,14 +25,17 @@
         and closes
         %em
           = @conference.call_for_paper.end_date.strftime('%A, %B %-d. %Y.')
-        That means you have only
-        - days = (@conference.call_for_paper.end_date - Date.today).to_i
-        %b
-          = pluralize(days, 'days')
+        - if @conference.cfp_open?
+          That means you have only
+          - days = @conference.call_for_paper.remaining_days
```
Comment (ChrisBr) [inline app/views/conference/_call_for_paper.html.haml]: You don't need to store this in a variable. Just 

```
= pluralize(@conference.call_for_paper.remaining_days, 'days')
```

should be ok!

File: app/views/proposal/_encouragement_text.html.haml [hunk 4]
```diff
@@ -6,18 +6,20 @@
...
     The submission period has begun
     %em
       = @conference.call_for_paper.start_date.strftime('%A, %B %-d. %Y')
     and closes
     %em
       = @conference.call_for_paper.end_date.strftime('%A, %B %-d. %Y.')
-    That means you have only
-    - days = (@conference.call_for_paper.end_date - Date.today).to_i
+    That means you have
+    - days = @conference.call_for_paper.remaining_days
```
Comment (ChrisBr) [inline app/views/proposal/_encouragement_text.html.haml]: Same here!

File: app/views/admin/call_for_papers/show.html.haml [hunk 5]
```diff
@@ -19,8 +19,12 @@
...
+            - days = @call_for_paper.remaining_days
+          - else
+            - days = 0
```
Comment (ChrisBr) [inline app/views/admin/call_for_papers/show.html.haml]: I'd move the zero check to the helper, so you don't need to check it in the view!

Sth like 

```
def remaining_days(date = Date.today)
  result = (self.end_date - date).to_i
  result > 0 ? result : 0
end
```

**Comment by ChrisBr** [comment]
Otherwise LGTM!

Squash the commits and we can :ship: it 
:clap:

**Comment by ChrisBr** [comment]
Awesome @raluka!
Thank you :bouquet:

**Pull request merged**